src/verify_cert.rs

```diff
@@ -14,14 +14,51 @@
 
 use crate::{
     cert::{self, Cert, EndEntityOrCa},
-    der, signed_data, subject_name, time, Error, SignatureAlgorithm, TrustAnchor,
+    der, signed_data, subject_name, time, CertRevocationList, Error, SignatureAlgorithm,
+    TrustAnchor,
 };
 
-pub(crate) fn build_chain(
-    required_eku_if_present: KeyPurposeId,
-    supported_sig_algs: &[&SignatureAlgorithm],
-    trust_anchors: &[TrustAnchor],
-    intermediate_certs: &[&[u8]],
+/// Specifies how much of a certificate chain is checked for revocation status.
+#[derive(Debug, PartialEq, Eq)]
+pub enum RevocationCheckDepth {
+    /// Only check the end entity certificate for revocation status.
+    EndEntity,
+    /// Check both the end entity certificate as well as any relevant intermediate certificates
+    /// for revocation status.
+    Chain,
+}
+
+/// A trait that can provide CRLs to use for revocation checking.
+pub trait CrlProvider<'a> {
+    /// A function that can be invoked with a [`Cert`] to optionally provide a [`CertRevocationList`]
+    /// to use to verify the certificate's revocation status.
+    /// TODO(@cpu): Requires making Cert exported. Maybe better to just pass an issuer?
+    fn crl_for_cert(&self, cert: &Cert) -> Option<&'a CertRevocationList<'a>>;
```

Comment:
The clippy error is related to the change to expose the `Cert` type.

I think I've come around to the idea of just sending the certificate serial & issuer through this trait but it's an open question whether we want to pass these as `untrusted::Input` and consider that part of our public API, or convert to some other representation to avoid that.